apierr/errors.go

```diff
@@ -73,10 +73,15 @@ func (apiError *APIError) IsTooManyRequests() bool {
 
 // IsRetriable returns true if error is retriable
 func (apiError *APIError) IsRetriable() bool {
+	return apiError.isRetriable(context.Background())
+}
+
+// isRetriable returns true if error is retriable
+func (apiError *APIError) isRetriable(ctx context.Context) bool {
 	// Handle transient errors for retries
 	for _, substring := range transientErrorStringMatches {
 		if strings.Contains(apiError.Message, substring) {
-			logger.Debugf("Attempting retry because of %#v", substring)
+			logger.Debugf(ctx, "Attempting retry because of %#v", substring)
```

Comment:
@nfx This is needed so that this log entry can be associated with log lines for the same request.